exe/wallet/jormungandr/Main.hs

```diff
@@ -77,197 +81,270 @@ import Cardano.Wallet.Unsafe
+-------------------------------------------------------------------------------}
 
-cliCommands :: String
-cliCommands = [here|
-  cardano-wallet launch [--network=STRING] [(--port=INT | --random-port)] [--backend-port=INT] [--state-dir=DIR] [(--quiet | --verbose )] --genesis-hash=HASH --genesis-block=PATH --node-config=PATH --node-secret=PATH
-  cardano-wallet serve  [--network=STRING] [(--port=INT | --random-port)] [--backend-port=INT] [--database=FILE] [(--quiet | --verbose )] --genesis-hash=HASH
-|]
+data LaunchArgs = LaunchArgs
+    { _listen :: Listen
+    , _nodePort :: Port "Node"
+    , _stateDir :: Maybe FilePath
+    , _verbosity :: Verbosity
+    , _jormungandrArgs :: JormungandrArgs
+    }
 
-cliOptions :: String
-cliOptions = [here|
-  --backend-port <INT>      port used for communicating with the backend node [default: 8081]
-  --genesis-block <FILE>    path to genesis block
-  --genesis-hash <HASH>     hash of genesis block
-  --node-config <FILE>      path to node configuration (general)
-  --node-secret <FILE>      path to node configuration (secret)
-|]
+data JormungandrArgs = JormungandrArgs
+    { _genesisBlock :: FilePath
+    , _nodeConfig :: FilePath
+    , _nodeSecret :: FilePath
+    }
 
-cliExamples :: String
-cliExamples = [here|
-  # Launch and monitor a wallet server and its associated chain producer
-  cardano-wallet launch \
-    --genesis-hash=dba597bee5f0987efbf56f6bd7f44c38158a7770d0cb28a26b5eca40613a7ebd \
-    --genesis-block=block0.bin \
-    --node-config config.yaml \
-    --backend-port 8081 \
-    --node-secret secret.yaml
+-- | cardano-wallet launch
+cmdLaunch
+    :: Mod CommandFields (IO ())
+cmdLaunch = command "launch" $ info (helper <*> cmd) $ mempty
+    <> progDesc "Launch and monitor a wallet server and its chain producers."
+  where
+    cmd = fmap exec $ LaunchArgs
+        <$> listenOption
+        <*> nodePortOption
+        <*> optional stateDirOption
+        <*> verbosityOption
+        <*> (JormungandrArgs
+            <$> genesisBlockOption
+            <*> nodeConfigOption
+            <*> nodeSecretOption)
+    exec (LaunchArgs listen nodePort stateDir verbosity jArgs) = do
+        block0H <- runGet getBlockId <$> BL.readFile (_genesisBlock jArgs)
+        execLaunch verbosity stateDir
+            [ commandJormungandr jArgs
+            , commandWalletServe block0H
+            ]
+      where
+        commandJormungandr (JormungandrArgs block0 nodeConfig nodeSecret) =
+            Command "jormungandr" arguments (return ()) Inherit
+          where
+            arguments = mconcat
+              [ [ "--genesis-block", block0 ]
+              , [ "--config", nodeConfig ]
+              , [ "--secret", nodeSecret ]
+              ]
 
-  # Start only a wallet server and connect it to an already existing chain producer
-  cardano-wallet serve --backend-port 8081
-|]
+        commandWalletServe block0H =
+            Command "cardano-wallet" arguments (return ()) Inherit
```

Comment:
I guess we have to be a bit careful here with naming of executables. (On my local machine, I changed this to `cardano-wallet-jormungandr` to facilitate testing, but of course that's a bit of a hack.)

I guess this problem will go away if we get rid of the `launch/merge` split.

Reply:
Yes, that's something I wanted to work on today. I think we can avoid hard-coding the name of the executable here and instead, do a dynamic lookup of the executable's path at runtime and use this as a command. This way, it'll work regardless how it is called.